src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -699,13 +699,25 @@ private IOperation CreateBoundObjectInitializerMemberOperation(BoundObjectInitia
         private ICollectionElementInitializerOperation CreateBoundCollectionElementInitializerOperation(BoundCollectionElementInitializer boundCollectionElementInitializer)
         {
             IMethodSymbol addMethod = boundCollectionElementInitializer.AddMethod;
-            Lazy<ImmutableArray<IOperation>> arguments = new Lazy<ImmutableArray<IOperation>>(() => boundCollectionElementInitializer.Arguments.SelectAsArray(n => Create(n)));
-            bool isDynamic = false;
+            Lazy<ImmutableArray<IArgumentOperation>> arguments = new Lazy<ImmutableArray<IArgumentOperation>>(() =>
+                DeriveArguments(
```

Comment:
>DeriveArguments [](start = 16, length = 15)

It looks like the information in BoundCollectionElementInitializer is actually getting pulled from BoundCall node and in some cases compiler can create BoundCall node even for erroneous cases. See how CreateBoundCallOperation helper is trying to intercept those, failure to do so is likely to cause a crash in DeriveArguments. I think we should be conservative and add similar guard here, come up with a test scenario would be good too.

Reply:
CreateBoundCall is trying to compensate for overload resolution failure, which is not a case we'll ever see in this method. See BindCollectionInitializerElementAddMethod in Binder_Expressions.cs. If overload resolution failed, we'd end up with a BoundCall, which will be properly handled by that create method.

---
In reply to: [187750757](https://github.com/dotnet/roslyn/pull/26781#discussion_r187750757) [](ancestors = 187750757)